game/scripts/vscripts/items/dagon.lua

```diff
@@ -42,6 +42,12 @@ function item_dagon:OnSpellStart()
     return
   end
 
+  -- If the target is an illusion, just kill it and don't do damage
```

Comment:
Should it also do damage so that fatal bonds does reflect the damage that it should reflect? merging either way because it's a strict improvement, but I kinda think it should still do the damage